lib/OpenQA/Worker/Settings.pm

```diff
@@ -81,6 +82,10 @@ sub _read_section ($cfg, $section, $out) {
     $out->{uc $_} = trim $cfg->val($section, $_) for $cfg->Parameters($section);
 }
 
+sub _read_section_keeping_default ($cfg, $section, $out) {
+    $out->{uc $_} //= trim $cfg->val($section, $_) for $cfg->Parameters($section);
```

Comment:
Could we change the name of the `$out` param?  why not `$settings`?

Reply:
I think `$out` is better as that shows it's the resulting outgoing value. `$setting` would be ambiguous as that could be the unparsed value